zerver/openapi/zulip.yaml

```diff
@@ -2755,7 +2755,44 @@ paths:
                                 Event sent to a user when [message flags][message-flags] are added
                                 to messages.
 
+                                This can reflect a direct user action, or can be the indirect
+                                consequence of another action. Whatever the cause, if there's a change
+                                in the set of message flags that the user has for a message, then an
+                                `update_message_flags` event will be sent with the change. Note
+                                that this applies when the user already had access to the message, and
+                                continues to have access to it. When a message newly appears or
+                                disappears, a [`message`][message-event] or
+                                [`delete_message`][message-delete] event is sent instead.
+
+                                Some examples of actions that trigger an `update_message_flags`
+                                event:
+
+                                - The `"starred"` flag is added when the user chooses to [star a
+                                  message](/help/star-a-message).
+                                - The `"read"` flag is added when the user marks messages as read by
+                                  scrolling through them, or uses [Mark all messages as read][all-read]
+                                  on a conversation.
+                                - The `"read"` flag is added when the user [mutes](/help/mute-a-user) a
+                                  message's sender.
+                                - The `"read"` flag is added after the user unsubscribes from a stream,
+                                  or messages are moved to a not-subscribed stream, provided the user
+                                  can still access the messages at all. Note a
+                                  [`delete_message`][message-delete] event is sent in the case the user
+                                  can no longer access the messages.
+
+                                In some cases, a change in message flags that's caused by another change
+                                may happen a short while after the original change, rather than
+                                simultaneously. For example, when messages that were unread are moved to
+                                a stream where the user is not subscribed, the resulting change in
+                                message flags (and the corresponding `update_message_flags` event with
+                                flag `"read"`) may happen later than the message move itself. The delay
+                                in that example is typically hundreds of milliseconds and can in rare
+                                cases be minutes or longer.
```

Comment:
I think the delay there is "typically at most a few hundred milliseconds" -- for streams that are not huge and an empty queue, which is in some sense the most typical state, it'll be <5ms.